### PR TITLE
Backport wathervaning

### PR DESCRIPTION
**Pull request by Charlie-Burge**

### CODE REVIEW

File: ArduCopter/mode.cpp [hunk 1]
```diff
@@ -827,3 +827,12 @@ uint16_t Mode::get_pilot_speed_dn()
 {
     return copter.get_pilot_speed_dn();
 }
+
+bool Mode::allows_weathervaning_auto(void) const
```
Comment (IamPete1) [inline ArduCopter/mode.cpp:831]: This function seems to be the same as `bool ModeAuto::allows_weathervaning(void) const` should just be able to remove this and swap in the other for the one user.

**Review by IamPete1 — DISMISSED** [review]
Looks OK, just that duplicate function. @Gone4Dirt should take a look tho since he did the work on this one.

File: ArduCopter/mode.cpp (hunk 1 above)
Comment (Gone4Dirt) [reply ArduCopter/mode.cpp:831]: You can remove the ModeAuto function if you want, that is a hang over from previous.  This one has to stay though as take off mode can't see the ModeAuto function, after some other changes to how the take off mission comand is handled.

File: ArduCopter/mode_auto.cpp [hunk 2]
```diff
@@ -148,6 +148,15 @@ bool ModeAuto::allows_arming(AP_Arming::Method method) const
     return (copter.g2.auto_options & (uint32_t)Options::AllowArming) != 0;
 };
 
+bool ModeAuto::allows_weathervaning(void) const
+{
```
Comment (Gone4Dirt) [inline ArduCopter/mode_auto.cpp:152]: If you want to this function can be removed.  It's just a hang over from changes I made last time I updated the PR to master.

**Review by Gone4Dirt — APPROVED** [review]
LGTM.  You remove the function in ModeAuto if you like.

Key differences for you to note over your previous version that has been in use:
- Time for weathervaning controller to activate after pilot has finished fiddling yaw control is now 2 sec (was 3 sec).
- Guided option bit and Auto option bit are now the 7th bit so you will need to update your master param set to reflect this.

File: ArduCopter/mode.cpp (hunk 1 above)
Comment (Charlie-Burge) [reply ArduCopter/mode.cpp:838]: have updated for that to be removed. This meant I needed to change the call in mode_auto.cpp to use Mode::allows_weathervaning_auto

**Review by Dan-Moss — APPROVED** [review]

File: ArduCopter/mode.cpp (hunk 1 above)
Comment (Gone4Dirt) [reply ArduCopter/mode.cpp:838]: Yeah, that's correct 👍

**Pull request merged**